transport/src/main/java/io/netty/channel/SingleThreadEventLoop.java

```diff
@@ -27,12 +28,24 @@
  */
 public abstract class SingleThreadEventLoop extends SingleThreadEventExecutor implements EventLoop {
 
+    private final Queue<Runnable> tailTasks;
+    private final LongCallable runTimeoutUpdater = new LongCallable() {
+        @Override
+        public void call(long aLong) {
+            runTimeout = aLong;
+        }
+    };
+    private boolean lastRunTimedOut;
+    private long runTimeout;
+
     protected SingleThreadEventLoop(EventLoopGroup parent, ThreadFactory threadFactory, boolean addTaskWakesUp) {
         super(parent, threadFactory, addTaskWakesUp);
+        tailTasks = newTaskQueue();
     }
 
     protected SingleThreadEventLoop(EventLoopGroup parent, Executor executor, boolean addTaskWakesUp) {
         super(parent, executor, addTaskWakesUp);
+        tailTasks = newTaskQueue();
```

Comment:
@NiteshKant do we need to do anything with the tailTasks if the user shutdown the `EventLoop` and there are still some in the queue ?